main.go

```diff
@@ -177,6 +179,11 @@ func main() {
 		os.Exit(1)
 	}
 
+	if region == "" {
```

Comment:
nit, this is ok to me, but since it's an new required arg, we should be extremely careful about this version especially for hot-patches. 
have we considered alternatively like only enable this sts header feature when it's specified or auto-discover region from ec2 metadata if not specified?

Reply:
I like the idea which only enable the sts header when region is provided. Will make an update.

Reply:
I have changed to check region. If empty, we use default client, otherwise we use enhanced client.